node/bft/src/helpers/proposal.rs

```diff
@@ -70,7 +70,7 @@ impl<N: Network> Proposal<N> {
             batch_header.transmission_ids().len() == transmissions.len(),
             "The transmission IDs do not match in the batch header and transmissions"
         );
-        for (a, b) in batch_header.transmission_ids().iter().zip_eq(transmissions.keys()) {
+        for (a, b) in batch_header.transmission_ids().iter().zip_eq(transmissions.keys().sorted()) {
```

Comment:
May need to revert this if we revert to IndexMap in the BatchHeader. We should decide on this before you waste time on fixing all of the snarkOS tests.